### ISSUE TITLE
FSDP fails on v2.20.3-1

### ISSUE THREAD

**Issue by alexdremov** [NVIDIA/nccl#1185]
Hi!

We recently updated out environment to the latest NCCL and encountered error with using fairscale FSDP algorithm. Training fails at (presumably) NCCL init

```
<host>:38:10588 [7] NCCL INFO Using non-device net plugin version 0
<host>:32:10592 [1] NCCL INFO Using non-device net plugin version 0
<host>:37:10586 [6] NCCL INFO Using non-device net plugin version 0
<host>:36:10591 [5] NCCL INFO Using non-device net plugin version 0
<host>:33:10589 [2] NCCL INFO Using non-device net plugin version 0
<host>:35:10587 [4] NCCL INFO Using non-device net plugin version 0
<host>:34:10590 [3] NCCL INFO Using non-device net plugin version 0
<host>:38:10588 [7] NCCL INFO Using network IBext_v8
<host>:31:10585 [0] NCCL INFO Using non-device net plugin version 0
<host>:32:10592 [1] NCCL INFO Using network IBext_v8
<host>:37:10586 [6] NCCL INFO Using network IBext_v8
<host>:36:10591 [5] NCCL INFO Using network IBext_v8
<host>:33:10589 [2] NCCL INFO Using network IBext_v8
<host>:35:10587 [4] NCCL INFO Using network IBext_v8
<host>:34:10590 [3] NCCL INFO Using network IBext_v8
<host>:31:10585 [0] NCCL INFO Using network IBext_v8
<host>:33:10589 [2] NCCL INFO bootstrapSplit: comm 0x7fd77cf70b90 parent 0xd809c70 rank 2 nranks 8 color -934961569 key 2 prev 1 next 3 - DONE
<host>:38:10588 [7] NCCL INFO bootstrapSplit: comm 0x7feb50f56ae0 parent 0xe1f0120 rank 7 nranks 8 color -934961569 key 7 prev 6 next 0 - DONE
<host>:33:10589 [2] NCCL INFO comm 0x7fd77cf70b90 rank 2 nranks 8 cudaDev 2 nvmlDev 2 busId 48000 commId 0xacb138a71bc342fc - Init START
<host>:37:10586 [6] NCCL INFO bootstrapSplit: comm 0x7fb210f52190 parent 0xc8fbf40 rank 6 nranks 8 color -934961569 key 6 prev 5 next 7 - DONE
<host>:38:10588 [7] NCCL INFO comm 0x7feb50f56ae0 rank 7 nranks 8 cudaDev 7 nvmlDev 7 busId d7000 commId 0xacb138a71bc342fc - Init START
<host>:31:10585 [0] NCCL INFO bootstrapSplit: comm 0x7f2a69e4add0 parent 0xcbcc450 rank 0 nranks 8 color -934961569 key 0 prev 7 next 1 - DONE
<host>:32:10592 [1] NCCL INFO bootstrapSplit: comm 0x7f0614f62120 parent 0xde17a70 rank 1 nranks 8 color -934961569 key 1 prev 0 next 2 - DONE
<host>:37:10586 [6] NCCL INFO comm 0x7fb210f52190 rank 6 nranks 8 cudaDev 6 nvmlDev 6 busId c2000 commId 0xacb138a71bc342fc - Init START
<host>:36:10591 [5] NCCL INFO bootstrapSplit: comm 0x7f69cd5f7670 parent 0xd15a870 rank 5 nranks 8 color -934961569 key 5 prev 4 next 6 - DONE
<host>:36:10591 [5] NCCL INFO comm 0x7f69cd5f7670 rank 5 nranks 8 cudaDev 5 nvmlDev 5 busId ae000 commId 0xacb138a71bc342fc - Init START
<host>:35:10587 [4] NCCL INFO bootstrapSplit: comm 0x7eff2da2f950 parent 0xdd76e60 rank 4 nranks 8 color -934961569 key 4 prev 3 next 5 - DONE
<host>:31:10585 [0] NCCL INFO comm 0x7f2a69e4add0 rank 0 nranks 8 cudaDev 0 nvmlDev 0 busId f000 commId 0xacb138a71bc342fc - Init START
<host>:35:10587 [4] NCCL INFO comm 0x7eff2da2f950 rank 4 nranks 8 cudaDev 4 nvmlDev 4 busId 87000 commId 0xacb138a71bc342fc - Init START
<host>:34:10590 [3] NCCL INFO bootstrapSplit: comm 0x7fa811a0d300 parent 0xc9e7200 rank 3 nranks 8 color -934961569 key 3 prev 2 next 4 - DONE
<host>:32:10592 [1] NCCL INFO comm 0x7f0614f62120 rank 1 nranks 8 cudaDev 1 nvmlDev 1 busId 34000 commId 0xacb138a71bc342fc - Init START
<host>:34:10590 [3] NCCL INFO comm 0x7fa811a0d300 rank 3 nranks 8 cudaDev 3 nvmlDev 3 busId 5a000 commId 0xacb138a71bc342fc - Init START
<host>:37:10586 [6] NCCL INFO Setting affinity for GPU 6 to ffffff00,00000000,00000000,ffffff00,00000000,00000000
<host>:33:10589 [2] NCCL INFO Setting affinity for GPU 2 to ffff,ff000000,00000000,0000ffff,ff000000
<host>:35:10587 [4] NCCL INFO Setting affinity for GPU 4 to ff,ffff0000,00000000,000000ff,ffff0000,00000000
<host>:36:10591 [5] NCCL INFO Setting affinity for GPU 5 to ffffff00,00000000,00000000,ffffff00,00000000,00000000
<host>:38:10588 [7] NCCL INFO Setting affinity for GPU 7 to ff,ffff0000,00000000,000000ff,ffff0000,00000000
<host>:32:10592 [1] NCCL INFO Setting affinity for GPU 1 to ffff,ff000000,00000000,0000ffff,ff000000
<host>:31:10585 [0] NCCL INFO Setting affinity for GPU 0 to ffffff,00000000,00000000,00ffffff
<host>:34:10590 [3] NCCL INFO Setting affinity for GPU 3 to ffffff,00000000,00000000,00ffffff
<host>:36:10591 [5] NCCL INFO comm 0x7f69cd5f7670 rank 5 nRanks 8 nNodes 1 localRanks 8 localRank 5 MNNVL 0
<host>:34:10590 [3] NCCL INFO comm 0x7fa811a0d300 rank 3 nRanks 8 nNodes 1 localRanks 8 localRank 3 MNNVL 0
<host>:35:10587 [4] NCCL INFO comm 0x7eff2da2f950 rank 4 nRanks 8 nNodes 1 localRanks 8 localRank 4 MNNVL 0
<host>:37:10586 [6] NCCL INFO comm 0x7fb210f52190 rank 6 nRanks 8 nNodes 1 localRanks 8 localRank 6 MNNVL 0
<host>:33:10589 [2] NCCL INFO comm 0x7fd77cf70b90 rank 2 nRanks 8 nNodes 1 localRanks 8 localRank 2 MNNVL 0
<host>:35:10587 [4] NCCL INFO Trees [0] 7/-1/-1->4->2 [1] 7/-1/-1->4->2 [2] 7/-1/-1->4->2 [3] 7/-1/-1->4->2 [4] 7/-1/-1->4->2 [5] 7/-1/-1->4->2 [6] 7/-1/-1->4->2 [7] 7/-1/-1->4->2 [8] 7/-1/-1->4->2 [9] 7/-1/-1->4->2 [10] 7/-1/-1->4->2 [11] 7/-1/-1->4->2 [12] 7/-1/-1->4->2 [13] 7/-1/-1->4->2 [14] 7/-1/-1->4->2 [15] 7/-1/-1->4->2 [16] 7/-1/-1->4->2 [17] 7/-1/-1->4->2 [18] 7/-1/-1->4->2 [19] 7/-1/-1->4->2 [20] 7/-1/-1->4->2 [21] 7/-1/-1->4->2 [22] 7/-1/-1->4->2 [23] 7/-1/-1->4->2
<host>:36:10591 [5] NCCL INFO Trees [0] 6/-1/-1->5->7 [1] 6/-1/-1->5->7 [2] 6/-1/-1->5->7 [3] 6/-1/-1->5->7 [4] 6/-1/-1->5->7 [5] 6/-1/-1->5->7 [6] 6/-1/-1->5->7 [7] 6/-1/-1->5->7 [8] 6/-1/-1->5->7 [9] 6/-1/-1->5->7 [10] 6/-1/-1->5->7 [11] 6/-1/-1->5->7 [12] 6/-1/-1->5->7 [13] 6/-1/-1->5->7 [14] 6/-1/-1->5->7 [15] 6/-1/-1->5->7 [16] 6/-1/-1->5->7 [17] 6/-1/-1->5->7 [18] 6/-1/-1->5->7 [19] 6/-1/-1->5->7 [20] 6/-1/-1->5->7 [21] 6/-1/-1->5->7 [22] 6/-1/-1->5->7 [23] 6/-1/-1->5->7
<host>:32:10592 [1] NCCL INFO comm 0x7f0614f62120 rank 1 nRanks 8 nNodes 1 localRanks 8 localRank 1 MNNVL 0
<host>:31:10585 [0] NCCL INFO comm 0x7f2a69e4add0 rank 0 nRanks 8 nNodes 1 localRanks 8 localRank 0 MNNVL 0
<host>:38:10588 [7] NCCL INFO comm 0x7feb50f56ae0 rank 7 nRanks 8 nNodes 1 localRanks 8 localRank 7 MNNVL 0
<host>:34:10590 [3] NCCL INFO Trees [0] 1/-1/-1->3->0 [1] 1/-1/-1->3->0 [2] 1/-1/-1->3->0 [3] 1/-1/-1->3->0 [4] 1/-1/-1->3->0 [5] 1/-1/-1->3->0 [6] 1/-1/-1->3->0 [7] 1/-1/-1->3->0 [8] 1/-1/-1->3->0 [9] 1/-1/-1->3->0 [10] 1/-1/-1->3->0 [11] 1/-1/-1->3->0 [12] 1/-1/-1->3->0 [13] 1/-1/-1->3->0 [14] 1/-1/-1->3->0 [15] 1/-1/-1->3->0 [16] 1/-1/-1->3->0 [17] 1/-1/-1->3->0 [18] 1/-1/-1->3->0 [19] 1/-1/-1->3->0 [20] 1/-1/-1->3->0 [21] 1/-1/-1->3->0 [22] 1/-1/-1->3->0 [23] 1/-1/-1->3->0
<host>:35:10587 [4] NCCL INFO P2P Chunksize set to 524288
<host>:36:10591 [5] NCCL INFO P2P Chunksize set to 524288
<host>:33:10589 [2] NCCL INFO Trees [0] 4/-1/-1->2->1 [1] 4/-1/-1->2->1 [2] 4/-1/-1->2->1 [3] 4/-1/-1->2->1 [4] 4/-1/-1->2->1 [5] 4/-1/-1->2->1 [6] 4/-1/-1->2->1 [7] 4/-1/-1->2->1 [8] 4/-1/-1->2->1 [9] 4/-1/-1->2->1 [10] 4/-1/-1->2->1 [11] 4/-1/-1->2->1 [12] 4/-1/-1->2->1 [13] 4/-1/-1->2->1 [14] 4/-1/-1->2->1 [15] 4/-1/-1->2->1 [16] 4/-1/-1->2->1 [17] 4/-1/-1->2->1 [18] 4/-1/-1->2->1 [19] 4/-1/-1->2->1 [20] 4/-1/-1->2->1 [21] 4/-1/-1->2->1 [22] 4/-1/-1->2->1 [23] 4/-1/-1->2->1
<host>:37:10586 [6] NCCL INFO Trees [0] -1/-1/-1->6->5 [1] -1/-1/-1->6->5 [2] -1/-1/-1->6->5 [3] -1/-1/-1->6->5 [4] -1/-1/-1->6->5 [5] -1/-1/-1->6->5 [6] -1/-1/-1->6->5 [7] -1/-1/-1->6->5 [8] -1/-1/-1->6->5 [9] -1/-1/-1->6->5 [10] -1/-1/-1->6->5 [11] -1/-1/-1->6->5 [12] -1/-1/-1->6->5 [13] -1/-1/-1->6->5 [14] -1/-1/-1->6->5 [15] -1/-1/-1->6->5 [16] -1/-1/-1->6->5 [17] -1/-1/-1->6->5 [18] -1/-1/-1->6->5 [19] -1/-1/-1->6->5 [20] -1/-1/-1->6->5 [21] -1/-1/-1->6->5 [22] -1/-1/-1->6->5 [23] -1/-1/-1->6->5
<host>:31:10585 [0] NCCL INFO Channel 00/24 :    0   3   1   2   4   7   5   6
<host>:32:10592 [1] NCCL INFO Trees [0] 2/-1/-1->1->3 [1] 2/-1/-1->1->3 [2] 2/-1/-1->1->3 [3] 2/-1/-1->1->3 [4] 2/-1/-1->1->3 [5] 2/-1/-1->1->3 [6] 2/-1/-1->1->3 [7] 2/-1/-1->1->3 [8] 2/-1/-1->1->3 [9] 2/-1/-1->1->3 [10] 2/-1/-1->1->3 [11] 2/-1/-1->1->3 [12] 2/-1/-1->1->3 [13] 2/-1/-1->1->3 [14] 2/-1/-1->1->3 [15] 2/-1/-1->1->3 [16] 2/-1/-1->1->3 [17] 2/-1/-1->1->3 [18] 2/-1/-1->1->3 [19] 2/-1/-1->1->3 [20] 2/-1/-1->1->3 [21] 2/-1/-1->1->3 [22] 2/-1/-1->1->3 [23] 2/-1/-1->1->3
<host>:34:10590 [3] NCCL INFO P2P Chunksize set to 524288
<host>:33:10589 [2] NCCL INFO P2P Chunksize set to 524288
<host>:37:10586 [6] NCCL INFO P2P Chunksize set to 524288
<host>:38:10588 [7] NCCL INFO Trees [0] 5/-1/-1->7->4 [1] 5/-1/-1->7->4 [2] 5/-1/-1->7->4 [3] 5/-1/-1->7->4 [4] 5/-1/-1->7->4 [5] 5/-1/-1->7->4 [6] 5/-1/-1->7->4 [7] 5/-1/-1->7->4 [8] 5/-1/-1->7->4 [9] 5/-1/-1->7->4 [10] 5/-1/-1->7->4 [11] 5/-1/-1->7->4 [12] 5/-1/-1->7->4 [13] 5/-1/-1->7->4 [14] 5/-1/-1->7->4 [15] 5/-1/-1->7->4 [16] 5/-1/-1->7->4 [17] 5/-1/-1->7->4 [18] 5/-1/-1->7->4 [19] 5/-1/-1->7->4 [20] 5/-1/-1->7->4 [21] 5/-1/-1->7->4 [22] 5/-1/-1->7->4 [23] 5/-1/-1->7->4
<host>:31:10585 [0] NCCL INFO Channel 01/24 :    0   3   1   2   4   7   5   6
<host>:32:10592 [1] NCCL INFO P2P Chunksize set to 524288
<host>:38:10588 [7] NCCL INFO P2P Chunksize set to 524288
<host>:31:10585 [0] NCCL INFO Channel 02/24 :    0   3   1   2   4   7   5   6
<host>:31:10585 [0] NCCL INFO Channel 03/24 :    0   3   1   2   4   7   5   6
<host>:31:10585 [0] NCCL INFO Channel 04/24 :    0   3   1   2   4   7   5   6
<host>:31:10585 [0] NCCL INFO Channel 05/24 :    0   3   1   2   4   7   5   6
<host>:31:10585 [0] NCCL INFO Channel 06/24 :    0   3   1   2   4   7   5   6
<host>:31:10585 [0] NCCL INFO Channel 07/24 :    0   3   1   2   4   7   5   6
<host>:31:10585 [0] NCCL INFO Channel 08/24 :    0   3   1   2   4   7   5   6
<host>:31:10585 [0] NCCL INFO Channel 09/24 :    0   3   1   2   4   7   5   6
<host>:31:10585 [0] NCCL INFO Channel 10/24 :    0   3   1   2   4   7   5   6
<host>:31:10585 [0] NCCL INFO Channel 11/24 :    0   3   1   2   4   7   5   6
<host>:31:10585 [0] NCCL INFO Channel 12/24 :    0   3   1   2   4   7   5   6
<host>:31:10585 [0] NCCL INFO Channel 13/24 :    0   3   1   2   4   7   5   6
<host>:31:10585 [0] NCCL INFO Channel 14/24 :    0   3   1   2   4   7   5   6
<host>:31:10585 [0] NCCL INFO Channel 15/24 :    0   3   1   2   4   7   5   6
<host>:31:10585 [0] NCCL INFO Channel 16/24 :    0   3   1   2   4   7   5   6
<host>:31:10585 [0] NCCL INFO Channel 17/24 :    0   3   1   2   4   7   5   6
<host>:31:10585 [0] NCCL INFO Channel 18/24 :    0   3   1   2   4   7   5   6
<host>:31:10585 [0] NCCL INFO Channel 19/24 :    0   3   1   2   4   7   5   6
<host>:31:10585 [0] NCCL INFO Channel 20/24 :    0   3   1   2   4   7   5   6
<host>:31:10585 [0] NCCL INFO Channel 21/24 :    0   3   1   2   4   7   5   6
<host>:31:10585 [0] NCCL INFO Channel 22/24 :    0   3   1   2   4   7   5   6
<host>:31:10585 [0] NCCL INFO Channel 23/24 :    0   3   1   2   4   7   5   6
<host>:31:10585 [0] NCCL INFO Trees [0] 3/-1/-1->0->-1 [1] 3/-1/-1->0->-1 [2] 3/-1/-1->0->-1 [3] 3/-1/-1->0->-1 [4] 3/-1/-1->0->-1 [5] 3/-1/-1->0->-1 [6] 3/-1/-1->0->-1 [7] 3/-1/-1->0->-1 [8] 3/-1/-1->0->-1 [9] 3/-1/-1->0->-1 [10] 3/-1/-1->0->-1 [11] 3/-1/-1->0->-1 [12] 3/-1/-1->0->-1 [13] 3/-1/-1->0->-1 [14] 3/-1/-1->0->-1 [15] 3/-1/-1->0->-1 [16] 3/-1/-1->0->-1 [17] 3/-1/-1->0->-1 [18] 3/-1/-1->0->-1 [19] 3/-1/-1->0->-1 [20] 3/-1/-1->0->-1 [21] 3/-1/-1->0->-1 [22] 3/-1/-1->0->-1 [23] 3/-1/-1->0->-1
<host>:31:10585 [0] NCCL INFO P2P Chunksize set to 524288

<host>:36:10591 [5] misc/ipcsocket.cc:57 NCCL WARN UDS: Binding to socket /tmp/nccl-socket-5-fb35dc94810b8fde failed : Address already in use (98)
<host>:36:10591 [5] NCCL INFO proxy.cc:1697 -> 2
<host>:36:10591 [5] NCCL INFO init.cc:1210 -> 2
<host>:36:10591 [5] NCCL INFO init.cc:1500 -> 2
<host>:36:10591 [5] NCCL INFO group.cc:64 -> 2 [Async thread]
<host>:36:10573 [5] NCCL INFO group.cc:418 -> 2
<host>:36:10573 [5] NCCL INFO group.cc:95 -> 2

<host>:33:10589 [2] misc/ipcsocket.cc:57 NCCL WARN UDS: Binding to socket /tmp/nccl-socket-2-e647f8d8e15809a1 failed : Address already in use (98)

<host>:38:10588 [7] misc/ipcsocket.cc:57 NCCL WARN UDS: Binding to socket /tmp/nccl-socket-7-8ed6cddae5ff6fdc failed : Address already in use (98)
<host>:33:10589 [2] NCCL INFO proxy.cc:1697 -> 2
<host>:33:10589 [2] NCCL INFO init.cc:1210 -> 2
<host>:38:10588 [7] NCCL INFO proxy.cc:1697 -> 2
<host>:38:10588 [7] NCCL INFO init.cc:1210 -> 2
<host>:33:10589 [2] NCCL INFO init.cc:1500 -> 2
<host>:33:10589 [2] NCCL INFO group.cc:64 -> 2 [Async thread]
<host>:38:10588 [7] NCCL INFO init.cc:1500 -> 2
<host>:38:10588 [7] NCCL INFO group.cc:64 -> 2 [Async thread]
<host>:33:10546 [2] NCCL INFO group.cc:418 -> 2
<host>:33:10546 [2] NCCL INFO group.cc:95 -> 2
<host>:38:10564 [7] NCCL INFO group.cc:418 -> 2
<host>:38:10564 [7] NCCL INFO group.cc:95 -> 2

<host>:31:10585 [0] misc/ipcsocket.cc:57 NCCL WARN UDS: Binding to socket /tmp/nccl-socket-0-6ba1fce36bab2ca3 failed : Address already in use (98)
<host>:31:10585 [0] NCCL INFO proxy.cc:1697 -> 2

<host>:32:10592 [1] misc/ipcsocket.cc:57 NCCL WARN UDS: Binding to socket /tmp/nccl-socket-1-48533f141101b1e2 failed : Address already in use (98)
<host>:31:10585 [0] NCCL INFO init.cc:1210 -> 2
<host>:32:10592 [1] NCCL INFO proxy.cc:1697 -> 2
<host>:32:10592 [1] NCCL INFO init.cc:1210 -> 2
<host>:31:10585 [0] NCCL INFO init.cc:1500 -> 2
<host>:31:10585 [0] NCCL INFO group.cc:64 -> 2 [Async thread]
<host>:32:10592 [1] NCCL INFO init.cc:1500 -> 2
<host>:32:10592 [1] NCCL INFO group.cc:64 -> 2 [Async thread]
<host>:31:10528 [0] NCCL INFO group.cc:418 -> 2
<host>:31:10528 [0] NCCL INFO group.cc:95 -> 2
<host>:32:10577 [1] NCCL INFO group.cc:418 -> 2
<host>:32:10577 [1] NCCL INFO group.cc:95 -> 2

<host>:37:10586 [6] misc/ipcsocket.cc:57 NCCL WARN UDS: Binding to socket /tmp/nccl-socket-6-992a96595161e79d failed : Address already in use (98)
<host>:37:10586 [6] NCCL INFO proxy.cc:1697 -> 2
<host>:37:10586 [6] NCCL INFO init.cc:1210 -> 2
<host>:37:10586 [6] NCCL INFO init.cc:1500 -> 2
<host>:37:10586 [6] NCCL INFO group.cc:64 -> 2 [Async thread]
<host>:37:10542 [6] NCCL INFO group.cc:418 -> 2
<host>:37:10542 [6] NCCL INFO group.cc:95 -> 2

<host>:34:10590 [3] misc/ipcsocket.cc:57 NCCL WARN UDS: Binding to socket /tmp/nccl-socket-3-edf2d53f0ced91e0 failed : Address already in use (98)
<host>:34:10590 [3] NCCL INFO proxy.cc:1697 -> 2
<host>:34:10590 [3] NCCL INFO init.cc:1210 -> 2
<host>:34:10590 [3] NCCL INFO init.cc:1500 -> 2
<host>:34:10590 [3] NCCL INFO group.cc:64 -> 2 [Async thread]
<host>:34:10558 [3] NCCL INFO group.cc:418 -> 2

<host>:35:10587 [4] misc/ipcsocket.cc:57 NCCL WARN UDS: Binding to socket /tmp/nccl-socket-4-1e849a63dbb50a9f failed : Address already in use (98)
<host>:34:10558 [3] NCCL INFO group.cc:95 -> 2
<host>:35:10587 [4] NCCL INFO proxy.cc:1697 -> 2
<host>:35:10587 [4] NCCL INFO init.cc:1210 -> 2
<host>:35:10587 [4] NCCL INFO init.cc:1500 -> 2
<host>:35:10587 [4] NCCL INFO group.cc:64 -> 2 [Async thread]
<host>:35:10524 [4] NCCL INFO group.cc:418 -> 2
<host>:35:10524 [4] NCCL INFO group.cc:95 -> 2
```

```
torch.distributed.DistBackendError: NCCL error in: /workspace/pytorch/torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:1691, unhandled system error (run with NCCL_DEBUG=INFO for details), NCCL version 2.20.3
ncclSystemError: System call (e.g. socket, malloc) or external library call failed or device error. 
Last error:
UDS: Binding to socket /tmp/nccl-socket-3-edf2d53f0ced91e0 failed : Address already in use (98)
```

The error is not reproduced on `v2.19.4-1`, so we will downgrade our setup for now

**Comment by AddyLaddy** [comment]
Ok thanks for this report, the issue is being provoked by `ncclCommSplit()` being called multiple times with the same ranks and color. We have a fix that we'll endeavor to release soon.

**Comment by alexdremov** [comment]
> Ok thanks for this report, the issue is being provoked by `ncclCommSplit()` being called multiple times with the same ranks and color. We have a fix that we'll endeavor to release soon.

Thanks!

Will look forward to it. Could you please ping me when the fix is ready so that I can test a new build and close the issue

**Comment by osayamenja** [comment]
I have the same issue as well with version `2.20.3`, looking forward to the fix.

**Comment by sjeaugey** [comment]
Fixed in 2.20.5. Closing.

**Comment by alexdremov** [comment]
> Fixed in 2.20.5. Closing.

Thanks!
Will check it out shortly
